Replace cloudscribe.HtmlAgilityPack with official HtmlAgilityPack to resolve System.Net.Http vulnerability

## Problem

Projects generated from cloudscribe templates show a persistent `NU1903` vulnerability warning during build:

```
warning NU1903: Package 'System.Net.Http' 4.3.0 has a known high severity vulnerability, https://github.com/advisories/GHSA-7jgj-8wvc-jh57
```

### Root Cause

The vulnerability originates from the following dependency chain:

```
cloudscribe.Web.Common 10.0.0
  ↓
cloudscribe.HtmlAgilityPack 1.0.1
  ↓
NETStandard.Library 1.6.1
  ↓
System.Net.Http 4.3.0 (VULNERABLE)
```

**cloudscribe.HtmlAgilityPack 1.0.1** is an old .NET Core port from 2017 that depends on `NETStandard.Library 1.6.1`, which brings in the vulnerable `System.Net.Http 4.3.0`.

The explicit override in `cloudscribe.Web.Common.csproj` (line 47) that references `System.Net.Http 4.3.4` does not eliminate the warning because NuGet still detects the vulnerable version in the transitive dependency tree.

## Solution

Replace `cloudscribe.HtmlAgilityPack 1.0.1` with the official **HtmlAgilityPack** NuGet package.

### Benefits
- The official package supports .NET Standard 2.0+ and .NET 5+
- No longer depends on the old `NETStandard.Library 1.6.1`
- Actively maintained by the original HtmlAgilityPack team
- Eliminates the vulnerability warning completely

### Scope of Changes

**1. Package Reference** (1 location):
- File: `src/cloudscribe.Web.Common/cloudscribe.Web.Common.csproj:42`
- Change:
  ```xml
  <!-- OLD -->
  <PackageReference Include="cloudscribe.HtmlAgilityPack" Version="1.0.1" />
  
  <!-- NEW -->
  <PackageReference Include="HtmlAgilityPack" Version="1.11.71" />
  ```

**2. Using Statement** (1 location):
- File: `src/cloudscribe.Web.Common/Html/HtmlHelper.cs:1`
- Change:
  ```csharp
  // OLD
  using cloudscribe.HtmlAgilityPack;
  
  // NEW
  using HtmlAgilityPack;
  ```

**3. Remove Obsolete Override** (optional cleanup):
- File: `src/cloudscribe.Web.Common/cloudscribe.Web.Common.csproj:47`
- Remove (no longer needed):
  ```xml
  <PackageReference Include="System.Net.Http" Version="4.3.4" />
  ```

### Code Compatibility

✅ The official HtmlAgilityPack uses **identical API** - no code changes required beyond the namespace.

The `HtmlHelper.ConvertUrlsToAbsolute()` method uses:
- `HtmlDocument` class
- `LoadHtml()` method
- `DocumentNode.Descendants()` method
- `Save()` method

All of these exist in the official package with the same signatures.

### Affected Packages

This change will impact the following cloudscribe 10.0 packages (transitive dependencies):
- `cloudscribe.Web.Common`
- `cloudscribe.ContentUtils`
- `cloudscribe.SimpleContent.Web`

### Testing Checklist

- [ ] Build cloudscribe.sln successfully
- [ ] Run unit tests: `dotnet test tests/cloudscribe.Tests/cloudscribe.UnitTests.csproj`
- [ ] Verify `HtmlHelper.ConvertUrlsToAbsolute()` works correctly
- [ ] Build generated template project (MSSQL configuration)
- [ ] Confirm NU1903 warning no longer appears
- [ ] Test SimpleContent HTML content with relative image/link URLs

### References

- Advisory: [GHSA-7jgj-8wvc-jh57](https://github.com/advisories/GHSA-7jgj-8wvc-jh57)
- Official HtmlAgilityPack: https://www.nuget.org/packages/HtmlAgilityPack/
- Old port repository: https://github.com/joeaudette/cloudscribe.HtmlAgilityPak (archived)

--- a/src/cloudscribe.Web.Common/Html/HtmlHelper.cs
+++ b/src/cloudscribe.Web.Common/Html/HtmlHelper.cs
@@ -1,4 +1,4 @@
-﻿using cloudscribe.HtmlAgilityPack;
+﻿using HtmlAgilityPack;
 using System;
 using System.Collections.Generic;
 using System.IO;

--- a/src/cloudscribe.Web.Common/cloudscribe.Web.Common.csproj
+++ b/src/cloudscribe.Web.Common/cloudscribe.Web.Common.csproj
@@ -39,12 +39,11 @@
 
     <PackageReference Include="System.Text.Encodings.Web" Version="10.0.0" />
 
-    <PackageReference Include="cloudscribe.HtmlAgilityPack" Version="1.0.1" />
+    <PackageReference Include="HtmlAgilityPack" Version="1.11.71" />
     <PackageReference Include="Newtonsoft.Json" Version="13.0.3" />
 
     <!-- pulling up transitive deps to resolve security vulns -->
     <PackageReference Include="System.Text.RegularExpressions" Version="4.3.1" />
-    <PackageReference Include="System.Net.Http" Version="4.3.4" />
     
   </ItemGroup>
 

